### DISCUSSION TITLE
Cannot create a transaction of any kind (expenses, revenue or transfers) due to invalid type

### DISCUSSION THREAD

**Discussion by Entropy-9973** [firefly-iii/firefly-iii#8399] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have googled the error and looked with chatgpt with no avail.
I cannot seem to add any transaction and I get an invalid type (should be withdrawl, expenses etc...)

### Debug information

{
	"message": "The selected transactions.0.type is invalid.",
	"errors": {
		"transactions.0.type": [
			"The selected transactions.0.type is invalid."
		]
	}
}

### Expected behaviour

The expected behaviour would be for the POST request to work

**Comment by JC5** [comment]
Are you using the user interface to create a transaction, or are you talking to the API directly?

**Comment by Entropy-9973** [comment]
> Are you using the user interface to create a transaction, or are you talking to the API directly?

I'm using the interface, here is the payload from chrome


{
	"0": {
		"amount": "98",
		"book_date": "",
		"category_name": "",
		"date": "2024-01-15T16:55",
		"description": "test",
		"destination_id": null,
		"destination_name": "fdsfsd",
		"due_date": "",
		"external_url": "",
		"interest_date": "",
		"internal_reference": "",
		"invoice_date": "",
		"notes": "",
		"payment_date": "",
		"process_date": "",
		"source_id": null,
		"source_name": "fdf",
		"type": "invalid"
	}
}

**Comment by JC5** [comment]
You can't just random dump names into the form. What are you trying to accomplish exactly?

**Comment by Entropy-9973** [comment]
> You can't just random dump names into the form. What are you trying to accomplish exactly?

That was a quick example of the content of the payload.
I just want to create an expense transaction. I keep getting that error and in the payload, the type is set as invalid which seems to be the problem.

Let s say that I want to put a trip to Berlin as a transaction, I believe the type should be set to expense. it is set to invalid which is causing the error (I haven't looked into the source code but probably induced by a constraint in the DB)

**Reply by JC5** [reply]
Close. If you want to make an expense, you need to select an asset account as the source.
